add map and tweak responsiveness of its container

nickvisut opened the pull request:
# Description

Add Mapbox map to main page and . Also adds a handful of example soft stories as [Mapbox markers](https://docs.mapbox.com/help/troubleshooting/markers-vs-layers/#markers).

Closes #72

## Type of changes
- [ ] Bugfix
- [ ] Chore
- [x] New Feature

## Testing
- [ ] I added automated tests
- [x] I think tests are unnecessary

## How to test

Run app, open index page and observe that map of SF is showing. Compare to Figma design. Attempt to pan/zoom with mouse and touch across diff devices and/or browser view port sizes.

Please also refer to requirements in #72, also linked in the right panel under "Development".

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+import React, { useRef, useEffect } from "react";
+import mapboxgl, { LngLat, LngLatLike } from "mapbox-gl";
+import "mapbox-gl/dist/mapbox-gl.css";
+import { m } from "framer-motion";
```
Maybe remove the unnecessary import?

nickvisut replied:
Definitely!

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+];
+
+/*
+//lng: number = -122.437, lat: number = 37.75
```
If we do not need this, maybe delete it? :)

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+//lng: number = -122.437, lat: number = 37.75
+*/
+
+//const Map = ({ lng = -122.437, lat = 37.75 }: { lng: number; lat: number }) => {
```
The same here, if we do not need this, maybe delete it? :)

nickvisut replied:
cleaned up!

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+        container: mapContainerRef.current!,
+        style: "mapbox://styles/mapbox/standard",
+        center: [-122.437, 37.75],
+        //center: [lng, lat],
```
If that's unnecessary, maybe delete it? :)

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+      const map = mapRef.current;
+
+      map.on("load", () => {
+        console.log("loaded");
```
I would suggest removing console.logs, if that's ok with you :)

nickvisut replied:
of course!

mantuok commented on app/components/map.tsx:
```diff
@@ -0,0 +1,101 @@
+        const el = document.createElement("div");
+
+        const center = map.getCenter();
+        console.log(center);
```
The same here, I would suggest removing console.logs, if that's ok with you :)

nickvisut replied:
Thank you! Yes, I plan on removing them before I take this out of draft status. Good eyes (o:

mantuok commented on app/globals.css:
```diff
@@ -0,0 +1,46 @@
+.key-tsunami {
+  /* pink/400 */
+  background-color: #ed64a6;
+}
```
I'm reviewing the design right now, and it seems that the current map style needs a few adjustments. The map should have a shadow and slightly rounded corners. :)

nickvisut replied:
You're right! I may hold off on that temporarily b/c the design may change to go edge-to-edge. If not, though, will def add in the rounded corners. I did not notice the shadow, so thanks for pointing that out.

nickvisut replied:
I added both the rounded corners and drop shadow. There are some slight whitespace issues that cause scrollbars at breakpoints, but FYI I'm going to wait to see if edge-to-edge design wins out to avoid rework.

nickvisut replied:
@mantuok this change was done via this commit https://github.com/sfbrigade/datasci-earthquake/pull/88/commits/5c143fd1301802d96a77d809aaa770803e619391 in `page.tsx`

nickvisut commented:
@mantuok OK I polished this up, added some missing things based on your feedback, and pulled it out of draft. FYI I rebased on `master`, so you may need to clear out this branch locally to get the new version.

Furthermore, note that I adopted your naming convention for the mapbox token, so you can get rid of the other one locally if you still see it there.

nickvisut commented:
> @mantuok OK I polished this up, added some missing things based on your feedback, and pulled it out of draft. FYI I rebased on `master`, so you may need to clear out this branch locally to get the new version.
> 
> Furthermore, note that I adopted your naming convention for the mapbox token, so you can get rid of the other one locally if you still see it there.

FYI I just did the force push that rebases on `master`.

nickvisut commented on .env.example:
```diff
@@ -15,5 +15,5 @@ SECRET_KEY=default-secret-key_local # Replace with your actual secret key
 # Frontend Environment Variables
 NEXT_PUBLIC_API_URL=http://localhost:8000/api # The base URL for API calls to the backend
-NEXT_PUBLIC_MAPBOX_TOKEN=dummy-mapbox-token
+NEXT_PUBLIC_MAPBOX_TOKEN=dummy-mapbox-token # Allows access to Mapbox APIs; replace with your actual access token
```
added a comment here when I consolidated mapbox tokens @mantuok

nickvisut commented on README.md:
```diff
@@ -86,7 +86,8 @@ This project uses Docker and Docker Compose to run the application, which includ
 2.  **Access the Application**:
-    - The app is running at http://localhost:3000.
+
+    - The app is running at http://localhost:3000. Note that this may conflict with your local dev server. If so, one will be running on port 3000 and the other on port 3001.
```
added a note here based on our convo regarding local dev vs Docker front end @agennadi; it could probably be clarified further, but what do you think?

nickvisut replied:
cc @mantuok would like your feedback as well since you'll have a good read on how a new front end dev would read this

agennadi replied:
Thank you for adding this note, @nickvisut.  I am going to update the README and mention that there are two separate options for running the app: locally and in Docker.

nickvisut commented on README.md:
```diff
@@ -124,8 +126,8 @@ We use GitHub Secrets to store sensitive environment variables. A template `.env
-- **Backend Environment Variables**. These variables are used by the backend (e.g., FastAPI) to configure its behavior and to connect to the database and the frontend application.
-- **Frontend Environment Variables**. This section contains the base URL for API calls to the backend and `NODE_ENV` variable that determines in which environment the Node.js application is running.
+- **Backend Environment Variables**. These variables are used by the backend (i.e., FastAPI) to configure its behavior and to connect to the database and the frontend application.
+- **Frontend Environment Variables**. This section contains the base URL for API calls to the backend, `NODE_ENV` variable that determines in which environment the Node.js application is running, and the token needed to access Mapbox APIs.
```
added in a reference to our Mapbox token

nickvisut commented on README.md:
```diff
@@ -124,8 +126,8 @@ We use GitHub Secrets to store sensitive environment variables. A template `.env
 - **Postgres Environment Variables**. This section contains the credentials to connect to the PostgreSQL database, such as the username, password, and the name of the database.
-- **Backend Environment Variables**. These variables are used by the backend (e.g., FastAPI) to configure its behavior and to connect to the database and the frontend application.
-- **Frontend Environment Variables**. This section contains the base URL for API calls to the backend and `NODE_ENV` variable that determines in which environment the Node.js application is running.
+- **Backend Environment Variables**. These variables are used by the backend (i.e., FastAPI) to configure its behavior and to connect to the database and the frontend application.
```
I have no idea how this got changed tbh. Is there some crazy linter that edits abbreviations?

nickvisut commented on app/components/map.tsx:
```diff
@@ -0,0 +1,93 @@
+import mapboxgl, { LngLat } from "mapbox-gl";
+import "mapbox-gl/dist/mapbox-gl.css";
+
+const lookupCoordinates = {
```
temporary and for reference; will be replaced at some point by props being passed in

nickvisut commented on app/components/map.tsx:
```diff
@@ -0,0 +1,93 @@
+  },
+};
+
+const softStories = [
```
temporary mock data will be replaced at some point by props being passed in

nickvisut commented on app/components/map.tsx:
```diff
@@ -0,0 +1,93 @@
+      mapRef.current = new mapboxgl.Map({
+        container: mapContainerRef.current!,
+        style: "mapbox://styles/mapbox/standard",
+        center: [-122.437, 37.75],
```
these coordinate zoom settings are arbitrary and pulled from https://github.com/sfbrigade/support-sfusd/; we can update based on feedback from @mjjsf once live/deployed somewhere.

nickvisut commented on app/components/map.tsx:
```diff
@@ -0,0 +1,93 @@
+          anchor: "bottom",
+          element: el,
+          className: "marker",
+          draggable: true,
```
may want to disable this as it doesn't currently do anything, but it can be used to implement a potentially planned feature

nickvisut replied:
I disabled this in the follow-up PR (#95)

nickvisut commented on app/globals.css:
```diff
@@ -0,0 +1,46 @@
+.marker {
```
I'm unaware if there is a way to style markers without global CSS at this time, but if we find one, I would recommend migrating to it.

nickvisut replied:
that being said, only the address lookup marker and other non-WebGL/non-Layer elements will remain here after moving to Mapbox Layers, so not a priority

nickvisut replied:
Moved to #102

nickvisut commented on app/globals.css:
```diff
@@ -0,0 +1,46 @@
+  border-radius: 0;
+}
+
+.soft-story {
```
the soft story CSS should be replaced with Mapbox Layer styling in an upcoming PR (since we need to replace soft story Markers with Layers)

nickvisut commented on app/page.tsx:
```diff
@@ -1,6 +1,8 @@
 import SearchBar from "./components/search-bar";
 import Heading from "./components/heading";
+import Map from "./components/map";
+import "./globals.css";
```
see https://github.com/sfbrigade/datasci-earthquake/pull/88/files#r1855298668

nickvisut commented on app/page.tsx:
```diff
@@ -20,17 +22,24 @@ const Home = () => {
-        w={{ base: "base", xl: "xl" }}
-        h={{ base: "166px", md: "213px", xl: "261px" }}
+        w={{ base: "base", md: "741px", xl: "xl" }}
+        h={{ base: "323px", md: "411px", xl: "462px" }}
```
height was adjusted to match spec

nickvisut commented on app/page.tsx:
```diff
@@ -20,17 +22,24 @@ const Home = () => {
+          base: "23px 24px 27px 24px",
           md: "37px 27px 28px 26px",
-          xl: "50px 127px 40px 127px",
+          xl: "50px 128px 40px 127px",
```
padding numbers may not exactly match spec, but could be made moot since we may replace map layout with an edge-to-edge one (as per @mjjsf)

nickvisut replied:
Moved this to a new issue #97

nickvisut commented on app/page.tsx:
```diff
@@ -20,17 +22,24 @@ const Home = () => {
+          border="1px solid"
+          borderColor="grey.400"
+          borderRadius="8px"
+          boxShadow="0px 2px 4px -1px #0000000F, 0px 4px 6px -1px #0000001A"
```
Known TODO:
- [ ] need to pull this and other hardcoded values in via the theme file

nickvisut replied:
Moved this to a new issue #97

nickvisut commented on README.md:
```diff
@@ -104,9 +105,10 @@ To stop and shut down the application:
     - Stop all services.
     - Remove the containers, but it will **not** delete volumes (so the database data will persist).
-   
+
```
prettier autoformatting

mantuok reviewed (approved):
Everything works and good to go!
